userspace/libsinsp/container_info.cpp

```diff
@@ -17,49 +17,246 @@ limitations under the License.
 
 */
 
+#include <utility>
+
 #include "container_info.h"
 #include "sinsp.h"
 #include "sinsp_int.h"
 
-const sinsp_container_info::container_mount_info *sinsp_container_info::mount_by_idx(uint32_t idx) const
+sinsp_container_info::container_health_probe::container_health_probe()
 {
-	if (idx >= m_mounts.size())
+}
+
+sinsp_container_info::container_health_probe::container_health_probe(const container_health_probe &other)
+	: m_probe_type(other.m_probe_type),
+	  m_health_probe_exe(other.m_health_probe_exe),
+	  m_health_probe_args(other.m_health_probe_args),
+	  m_obj(other.m_obj)
+{
+}
+
+sinsp_container_info::container_health_probe::container_health_probe(container_health_probe &&other)
+	: m_probe_type(other.m_probe_type),
+	  m_health_probe_exe(std::move(other.m_health_probe_exe)),
+	  m_health_probe_args(std::move(other.m_health_probe_args)),
+	  m_obj(std::move(other.m_obj))
+{
+}
+
+sinsp_container_info::container_health_probe::~container_health_probe()
+{
+}
+
+sinsp_container_info::container_health_probe &sinsp_container_info::container_health_probe::operator=(const container_health_probe &other)
+{
+	m_probe_type = other.m_probe_type;
+	m_health_probe_exe = other.m_health_probe_exe;
+	m_health_probe_args = other.m_health_probe_args;
+	m_obj = other.m_obj;
+
+	return *this;
+}
```

Comment:
What do these methods do that can't be deduced by the compiler? They look pretty `= default` to me but I'm ready to be enlightened

Reply:
I needed all of them to get move semantics to work--I was trying to make appending to the list of probes efficient but if I recall the operator= method was = delete so I had to implement them.